### DISCUSSION TITLE
Wordpress - Passing User IDs

### DISCUSSION THREAD

**Discussion by gboxxx** [formbricks/formbricks#559] [General Requests]
Need some help passing Wordpress/Woocommerce user IDs, as this code is not working. 

```
function set_formbricks_user_id($user_login, $user) {
    ?>
    <script>
        formbricks.setUserId("<?php echo $user->ID; ?>");
    </script>
    <?php
}
add_action('wp_login', 'set_formbricks_user_id', 10, 2);

function logout_formbricks_user() {
    ?>
    <script>
        formbricks.logout();
    </script>
    <?php
}
add_action('wp_logout', 'logout_formbricks_user');
```

**Comment by mattinannt** [comment]
@gboxxx is this throwing an error or just silently just not setting the userId?

**Comment by gboxxx** [comment]
No error that I see!

On Mon, Jul 17, 2023 at 10:43 AM Matti Nannt ***@***.***>
wrote:

> @gboxxx <https://github.com/gboxxx> is this throwing an error or just
> silently just not setting the userId?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/formbricks/formbricks/discussions/559#discussioncomment-6469318>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXBYUUJFYANHQHTV2LV2A5DXQVMRNANCNFSM6AAAAAA2KQPX3Y>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Stay awesome,

[image: Company logo]

Vincent Termini VP, Marketing & Automation
Sacramento, CA
(916) 407-4884 | GrubBoxx.com

**Comment by mattinannt** [comment]
Sorry for taking so long to get back to you. Were you able to resolve the error?
The best option to go further to debug the issue is to make sure you use the current version of formbricks-js and enable debug mode. Currently the newest version is version `1.0.0`.
When embedding formbricks-js using the javascript snippet you can enable debug mode by using this snipped (notice `debug` is set to true inside the `formbricks.init()` call):
```js
<script type="text/javascript">
  !(function () {
    var t = document.createElement("script");
    (t.type = "text/javascript"),
      (t.async = !0),
      (t.src = "https://unpkg.com/@formbricks/js@^1.0.0/dist/index.umd.js");
    var e = document.getElementsByTagName("script")[0];
    e.parentNode.insertBefore(t, e),
      setTimeout(function () {
        window.formbricks = window.js;
        window.formbricks.init({
          environmentId: "your-environment-id",
          apiHost: "https://app.formbricks.com",
          debug: true,
        });
      }, 500);
  })();
</script>
```
When debug mode is on, you can see different logs inside your browsers javascript console can help you to see if a certain action is triggered or not. Can you please try this and check if formbricks get initialized properly and check if the change of the userId gets triggered?

When you move into production, make sure to remove the `debug: true,` again.
